rosidl_typesupport_opendds_cpp/include/rosidl_typesupport_opendds_cpp/service_type_support.h

```diff
@@ -18,31 +18,45 @@
 #include <stdint.h>
 #include <rmw/types.h>
 #include "rosidl_generator_c/service_type_support_struct.h"
+#include "dds/DCPS/Service_Participant.h"
+
```

Comment:
In general, avoid including implementation-specific headers in header files reduces unwanted dependencies. That's the reason we decided using opaque types (void*) here for OpenDDS objects to avoid potential problems. If this is properly verified as unnecessary, then I would suggest using references for the OpenDDS objects here.

Reply:
I don't think this is a problem.  We are not depending on any implementation details of rosidl_typesupport here, just the common OpenDDS headers that are available to all.  I will defer to others on ROS2 build / packaging system details.

The one change I would suggest would be to `#include` the specific OpenDDS headers that are needed instead of `Service_Participant.h`.

Reply:
In earlier discussions I've got an impression that passing OpenDDS types would be preferable, if it's easy. That would save us type casting / conversion. Should I go back and change all types to 'untyped'?

Reply:
No, at least not for now.  If you have a build locally that's compiling all of what should be compiled, that's probably a good sign that it can work.